bestla/bestla/bestla_parallel.h

```diff
@@ -30,6 +30,36 @@ using thread_func = std::function<void(int tid)>;
 class IThreading {
  public:
   explicit IThreading(int nthreads, bool supportPE) : mThreadNum(nthreads), isSupportPE(supportPE) {}
+
+  // equal to "for(int i=begin1;i<end1;i+=step1)"
+  void parallel_for_collapse(const int& begin1, const int& end1, const int& step1,
+                             const std::function<void(int)>& func) {
+    assert(end1 >= begin1 && step1 > 0);
+    parallel_for([&](int tidx) {
+      const int length = (end1 - begin1 + step1 - 1) / step1;
+      const int block_size = length / mThreadNum;
+      const int block_remain = length % mThreadNum;
+      const int offset = std::min(tidx, block_remain) + tidx * block_size;
```

Comment:
what does this `std::min(tidx, block_remain)` mean?

Reply:
The first block_remain threads will do `block_size+1` and the others will do `block_size` in the division. `offset` is the begin of each thread.